recipes/sol2/3.x.x/conanfile.py

```diff
@@ -0,0 +1,71 @@
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class Sol2Conan(ConanFile):
+    name = "sol2"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ThePhD/sol2"
+    description = "C++17 Lua bindings"
+    topics = ("conan", "lua", "c++", "bindings")
+    settings = "os", "compiler"
+    license = "MIT"
+    requires = ["lua/5.3.5"]
+    
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+    
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder
+        )
+        return self._cmake
+
+
+    def _has_support_for_cpp17(self):
+        supported_compilers = [("apple-clang", 10), ("clang", 6), ("gcc", 7), ("Visual Studio", 15.7)]
+        compiler, version = self.settings.compiler, tools.Version(self.settings.compiler.version)
+        return any(compiler == sc[0] and version >= sc[1] for sc in supported_compilers)
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17")
+        if not self._has_support_for_cpp17():
+            raise ConanInvalidConfiguration("sol2 requires C++17 or higher support standard."
+                                            " {} {} is not supported."
+                                            .format(self.settings.compiler,
+                                                    self.settings.compiler.version))
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
```

Comment:
```suggestion
```

As it's header-only, you don't need to build anything. Only calling CMake.configure() and CMake.install() is enough

Reply:
It does something, but I delete the result  (genrated pkg config and cmake files :-) 
so you are right 
just ask myself, when are such files to be added, and when to be deleted, I am not totally clear about that and do what most other do

Reply:
like you said, it generates build files only, as cmake and pkg config, but [they are not allowed](https://github.com/conan-io/conan-center-index/wiki/FAQ#why-are-cmake-findconfig-files-and-pkg-config-files-not-packaged). Thus, the build step here is useless, you should remove it.

You only can package cmake files from the author, when there is a specific macro, which is not supported from conan generators.